packages/ses/src/module-load.js

```diff
@@ -231,21 +214,26 @@ function* loadWithoutErrorAnnotation(
       return aliasRecord;
     }
 
-    // check if this redirect with an explicit compartment
-    if (staticModuleRecord.compartment !== undefined) {
-      // ensure expected record shape
-      if (staticModuleRecord.importMeta !== undefined) {
-        throw TypeError(
-          'Cannot redirect to an implicit record with a specified importMeta',
-        );
+    // A (legacy) module descriptor that describes a link to a module instance
+    // in a specified compartment.
+    if (
+      moduleDescriptor.compartment !== undefined &&
+      moduleDescriptor.specifier !== undefined
```

Comment:
The legacy module descriptors with { compartment, specifier } correspond to { namespace, compartment } descriptors in XS, where the namespace is a string, the specifier. XS also allows the namespace to be a literal namespace or an ordinary object. If it’s an ordinary object, XS virtualizes a module to export its properties. We do not support any of the non-string cases in “specifier descriptors”.

Reply:
No, we’ve no intention of proposing any shape of `{specifier, compartment}` module descriptor. The motivating cases are fully subsumed into `{namespace, compartment?}` descriptors.

Reply:
Resolving